src/project_schema/index.ts

```diff
@@ -2,12 +2,20 @@ import build from './build';
 import src from './src';
 
 import { FolderGenerator, TemplateGenerator } from 'src/generators';
-import { TemplateOptions } from 'src/template_options';
+import { TemplateOptions, PluginType, Language, srcExt } from 'src/template_options';
 
 
 export default new FolderGenerator<TemplateOptions>(context => context.options.id, [
-  new TemplateGenerator(require('./package.json.ejs'), 'package.json'),
+  new TemplateGenerator(require('./package.json.ejs'),
+                        'package.json',
+                        (ctx) => { ctx['extension'] = srcExt('{ext}')(ctx) }
+  ),
   new TemplateGenerator(require('./.gitignore.ejs'), '.gitignore'),
   new TemplateGenerator(require('./README.md.ejs'), 'README.md'),
+  context => {
+    if(context.options.pluginType === PluginType.Panel && context.options.language === Language.TypeScript) {

```

Comment:
`tsconfig` for panel is the same as for datasource
So I think we can check only for language

Reply:
@rozetko  but if only JavaScript required?